### ISSUE TITLE
Unfurl GitHub links is not working

### ISSUE THREAD

**Issue by laurentkempe** [integrations/microsoft-teams#28]
First, I would like to say that I ❤️ this new integration 👍 to the team that brought it 👏 

Everything is working for us except the Unfurl GitHub links.
I am signed in into my Github account, I subscribed to two repos in one Team channel and the Microsoft Teams app is installed in the Github repo and allowed for our repos.
When in the chat someone past a link to a pull request, nothing happen!

How can we troubleshoot that?

**Comment by ashokirla** [comment]
Thanks for the feedback @laurentkempe. For the url to unfurl, all you need is to signin and allow access to Teams app in GitHub for the repos. As you said you have done it all, we will get back to you if we need further diagnostics to resolve this issue.

Meanwhile can you try restarting teams. And also try sharing the url in a Team channel where our GitHub app is not installed. Also do you have any other 3rd party GitHub app installed in your Teams.

**Comment by gauravsaralMs** [comment]
@laurentkempe : thanks a lot for trying it out.
we will be happy to help with the unfurl issue.

just for clarity, you are posting the pull request link in a private chat or in a channel?

**Comment by DivyankaM** [comment]
Hi @laurentkempe,

Could you please help us with some information so that we can debug this and make this work for you -

1. Make sure that you have app installed for you personally as well in case you want to have the unfurling working for you in chat.
2. Can you try posting the link in a teams channel where our app is installed for you and tell us if it works there?
3. Make sure the GitHub app is installed in the repository where the Pull request is present. (link of which you want to unfurl)
4. Make sure the link is valid and you have access to that link (Try to open the link in GH portal).
5. Could you share the link to the PR url you are trying to unfurl, so that we can check the issue?

**Comment by holljen** [comment]
I have a similar issue. Unfurling works for public repo's but not for private repos.

**Comment by DivyankaM** [comment]
Hi @holljen 

Could you please check that the GitHub app is installed in the repository in which that link is present which you want to unfurl. Please let me know in case you still face any issues. Thanks.

**Comment by holljen** [comment]
I have installed it globally for our organisation (for all repos). Do I need to go in and add it separately for each repo in addition?

**Comment by holljen** [comment]
Update: Seems to be a crash with the other (old) GitHub app. Investigating.

**Comment by DivyankaM** [comment]
@holljen Sure Thanks for the update. You need to have only one GitHub app installed for you for unfurling to work properly. I would suggest you to uninstall all the GitHub apps installed for you and re-install our GitHub app. Please let me know if this works for you :)

**Comment by DivyankaM** [comment]
> I have installed it globally for our organisation (for all repos). Do I need to go in and add it separately for each repo in addition?

FYI, It is enough to install github app once for all repos. Installing the app separately for each repo is not required.

**Comment by holljen** [comment]
Yes it works now. It is just ultra-confusing because we have tons of teams and you have to uninstall it from each team individually. If the old one is left in just one place it bugs out and it is hard to understand why. We also had to chat with your integration after removing all the old integrations to make unfurling work.

**Comment by DivyankaM** [comment]
> Yes it works now. It is just ultra-confusing because we have tons of teams and you have to uninstall it from each team individually. If the old one is left in just one place it bugs out and it is hard to understand why. We also had to chat (send one link) with your integration after removing all the old integrations to make unfurling work.

Yes @holljen We can understand your concern. This is a limitation from Teams platform end. Thanks for understanding. I am glad it works for you now :)  cc: @ashokirla

**Comment by laurentkempe** [comment]
Thanks @ashokirla and @DivyankaM to get back so quickly!

I uninstalled the Teams addons and the Apps from Github repo. I uninstalled the old Github app from the Teams, I have access to.
I reinstalled the whole and no chance! Lucky you @holljen

**Comment by DivyankaM** [comment]
Hey @laurentkempe Could you please try posting a public url , say 'https://github.com/GitHub' on your channel where GitHub app is installed and let me know what response do you get? Please wait for few seconds before posting the url. Thanks.

**Comment by laurentkempe** [comment]
> Hey @laurentkempe Could you please try posting a public url , say 'https://github.com/GitHub' on your channel where GitHub app is installed and let me know what response do you get? Please wait for few seconds before posting the url. Thanks.

I did wait 1 minute! It doesn't work @DivyankaM 
I just thought we are using two factor authentication, could that be a problem?

![image](https://user-images.githubusercontent.com/272612/92925719-33f28100-f43b-11ea-8fe3-2a76f3d737bb.png)

**Comment by ashokirla** [comment]
@holljen We understand. We are following up with Teams platform to figure out any alternate solution. For now we will update the documentation to make it clear.

**Comment by ashokirla** [comment]
@laurentkempe Are you sure there is no other 3rd party app installed in any channels or in personal scope. Can we setup a call so that we can look into your setup.

**Comment by laurentkempe** [comment]
@ashokirla I just uninstalled everything again, double check again the 3rd party app, I could find that I had a webhook connector in the channel which I wanted to add the app to, which I also removed. I restarted Teams, reinstalled all and still not workin!
So, yeah having a call would be fantastic!

**Comment by ashokirla** [comment]
@laurentkempe Sure, I will reach out to you over email. We can schedule sometime next week.

**Comment by ashokirla** [comment]
Updating the issue for the recod:
We had a call with customer. The URL unfurling is not working for any url and for any other apps as well. Customer got the message policy to 'Allow URL unfurl' enabled by the admin. Customer will reinstall the app and restart Teams and try it out.

**Comment by laurentkempe** [comment]
I tried again @ashokirla to install it and it doesn't work 😭
